ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java

```diff
@@ -327,4 +328,75 @@ public static String getBuildDate(final Properties properties) {
         return properties.get("build.date").toString();
     }
 
+    /**
+     * parse a string to get absolute date
+     * @param diff format Yyear.Mmonth.Ddate.hhours.mminutes.sseconds e.g., "Y2016.M11.D10.h12"
+     * @return long time in ms
+     */
+    public static Date parseParamDate(final String diff) throws java.text.ParseException {
```

Comment:
It seems that since you already have all the information you need that you can just create the date object yourself, rather than converting over to a different format, and then using the parser.